habitat_baselines/rl/hrl/hierarchical_policy.py

```diff
@@ -0,0 +1,208 @@
+import os.path as osp
+
+import torch
+
+from habitat_baselines.common.baseline_registry import baseline_registry
+from habitat_baselines.common.logging import logger
+from habitat_baselines.rl.hrl.high_level_policy import (  # noqa: F401.
+    GtHighLevelPolicy,
+    HighLevelPolicy,
+)
+from habitat_baselines.rl.hrl.skills import (  # noqa: F401.
+    NavSkillPolicy,
+    NnSkillPolicy,
+    OracleNavPolicy,
+    PickSkillPolicy,
+    PlaceSkillPolicy,
+)
+from habitat_baselines.rl.ppo.policy import Policy
+from habitat_baselines.utils.common import get_num_actions
+
+
+@baseline_registry.register_policy
+class HierarchicalPolicy(Policy):
+    def __init__(
+        self, config, full_config, observation_space, action_space, num_envs
+    ):
+        super().__init__()
+
+        self._action_space = action_space
+        self._num_envs = num_envs
+
+        # Maps (skill idx -> skill)
+        self._skills = {}
+        self._name_to_idx = {}
+
+        for i, (skill_id, use_skill_name) in enumerate(
+            config.USE_SKILLS.items()
+        ):
+            if use_skill_name == "":
+                # Skip loading this skill if no name is provided
+                continue
+            skill_config = config.DEFINED_SKILLS[use_skill_name]
+
+            cls = eval(skill_config.skill_name)
+            skill_policy = cls.from_config(
+                skill_config, observation_space, action_space, self._num_envs
+            )
+            self._skills[i] = skill_policy
+            self._name_to_idx[skill_id] = i
+
+        self._call_high_level = torch.ones(self._num_envs)
+        self._cur_skills: torch.Tensor = torch.zeros(self._num_envs)
+
+        high_level_cls = eval(config.high_level_policy.name)
+        self._high_level_policy: HighLevelPolicy = high_level_cls(
+            config.high_level_policy,
+            osp.join(
+                full_config.TASK_CONFIG.TASK.TASK_SPEC_BASE_PATH,
+                full_config.TASK_CONFIG.TASK.TASK_SPEC + ".yaml",
+            ),
+            num_envs,
+            self._name_to_idx,
+        )
+        self._stop_action_idx = 0
+        found = False
+        for k in action_space:
+            if k == "REARRANGE_STOP":
```

Comment:
Should not use a hard coded string

Reply:
I see that typically actions are referrenced via strings, is there a standard way to specify an action name string and reference it?

Reply:
There does not seem to be a clean way to refer to an action currently. Changing this is out of scope, feel free to ignore this comment.